src/test/java/jp/co/yahoo/yosegi/message/formatter/json/TestJacksonArrayFormatter.java

```diff
@@ -0,0 +1,358 @@
+    public boolean isArray() throws IOException {
+      return true;
+    }
+
+    @Override
+    public boolean isMap() throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean isStruct() throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean hasParser( final int index ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean hasParser( final String key ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public Object toJavaObject() throws IOException {
+      return null;
+    }
+
+  }
+
+  private class TestPrimitiveParser implements IParser {
+
+    private final boolean isArray;
+
+    public TestPrimitiveParser( final boolean isArray ) {
+      this.isArray = isArray;
+    }
+
+    @Override
+    public PrimitiveObject get(final String key ) throws IOException {
+      if ( "key1".equals( key ) ) {
+        return new StringObj( "a" );
+      } else if ( "key2".equals( key ) ) {
+        return new StringObj( "b" );
+      } else if ( "key3".equals( key ) ) {
+        return new StringObj( "c" );
+      }
+      return null;
+    }
+
+    @Override
+    public PrimitiveObject get( final int index ) throws IOException {
+      if ( index == 0 ) {
+        return new StringObj( "a" );
+      } else if ( index == 1 ) {
+        return new StringObj( "b" );
+      } else if ( index == 2 ) {
+        return new StringObj( "c" );
+      }
+      return null;
+    }
+
+    @Override
+    public IParser getParser( final String key ) throws IOException {
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public IParser getParser( final int index ) throws IOException {
+      return new JacksonNullParser();
+    }
+
+    @Override
+    public String[] getAllKey() throws IOException {
+      return new String[]{ "key1" , "key2" , "key3" };
+    }
+
+    @Override
+    public boolean containsKey( final String key ) throws IOException {
+      if ( "key1".equals( key ) ) {
+        return true;
+      } else if ( "key2".equals( key ) ) {
+        return true;
+      } else if ( "key3".equals( key ) ) {
+        return true;
+      }
+      return false;
+    }
+
+    @Override
+    public int size() throws IOException {
+      return 3;
+    }
+
+    @Override
+    public boolean isArray() throws IOException {
+      return isArray;
+    }
+
+    @Override
+    public boolean isMap() throws IOException {
+      return ! isArray;
+    }
+
+    @Override
+    public boolean isStruct() throws IOException {
+      return ! isArray;
+    }
+
+    @Override
+    public boolean hasParser( final int index ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean hasParser( final String key ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public Object toJavaObject() throws IOException {
+      return null;
+    }
+
+  }
+
+  @Test
+  public void T_write_1() throws IOException {
```

Comment:
Please write the test method name is to describe what is wanted to in the test instead of using number (it makes vague one).